### PR TITLE
build_display filter also allow a JObject input.

### PR DESCRIPTION
**Pull request by jtkech**
@sebastienros 

Then, doing this i could do the following 2 things in the `LandingPage__Services` template.

    {% for item in Model.ContentItems %}
        {% display item | build_display: "Summary" %}
    {% endfor %}

    // this also works because we now allow JObject input
    {% for item in Model.BagPart.Content.ContentItems %}
        {% display item | build_display: "Summary" %}
    {% endfor %}

**So, it seems that we don't need anymore to put `.ContentItems` directly in the `BagPart` model**.

**Update**: Just saw that other filters use a `ContentItem` input. Maybe good to do the same to allow also a `JObject`. I tried it with `display_text` filter, it works. Let me know.

### CODE REVIEW

**Comment by sebastienros** [comment]
What about changing the query module to return the clr object if we check "return document"?
If a user really wants a JObject, we could have a filter that just creates a JObject from a json string.

**Comment by jtkech** [comment]
> What about changing the query module to return the clr object if we check "return document"?

Yes, good idea, as you want for the query module.

But maybe still useful to be able to diplay from a `JObject` **as in my example where there is no query**.

1. We can't access `Model.BagPart.ContentItems` which are `ContentItems`.

2. We can access `Model.BagPart.Content.ContentItems` but there are `JObject`.

3. That's why we put them in `Model.ContentItems`.

- **But with this change we could display items by using 2. without needing 3.** This would prevent us from having to change other view models for the same reason.

> If a user really wants a JObject, we could have a filter that just creates a JObject from a json string.

Ok, but if related to your 1st suggestion, maybe you also mean `creates a JObject from a clr object`.

**Comment by sebastienros** [comment]
New requirement for SQL:
1- Return raw data as JObject
2- Return document content as JObject (the query needs to return ids)
3- Return document as Content Item (the query needs to return ids)

Ideally we should merge 2 and 3 by retrieving the Document's type and have the JObject deserialized into it's target class, but I am not sure the API lets you do that. If you manage to do it then go for it.

**Comment by sebastienros** [comment]
Note that with these options you can use radio buttons, even if we have just two options, at least to be explicit about what the result is.

**Comment by jtkech** [comment]
@sebastienros sorry i'm lost.

- This PR is not only related to `Query` because there are other contexts where we have a `JObject` but not a `ContentItem`. Remember when had to copy `Model.BagPart.ContentItems` to `Model.ContentItems`.

- So this PR is to also allow to use `build_display` from a `JObject`. And i saw some other filters that only allow a `ContentItem` input and where we could do the same, e.g `display_text`.

- Note: When i do `jObject.ToObject<ContentItem>()`, the right `ContentItemConverter` is used.

---

- Related to sql query, working on #1178, the sql query i used was already returning a `JObject` (even if Return document option is unchecked) as you said above in 1. and 2., so i don't understand.

- Then in #1178, i could get it working by checking the option and applying the changes in this PR.

- Finally i understand we could mix 2. and 3. so we don't need to do the conversion in the filter. But, as said above, we had and we could still have other contexts where we would need to do it. Note: as i remember the only specific filter input types are `Shape` and `ContentItem` ...

File: src/OrchardCore.Modules/OrchardCore.Contents/Filters/BuildDisplayFilter.cs [hunk 1]
```diff
@@ -13,9 +14,19 @@ public class BuildDisplayFilter : ILiquidFilter
...
+            {
+                contentItem = null;
+
+                if (obj is JObject jObject)
+                {
+                    contentItem = jObject.ToObject<ContentItem>();
+                }
+            }
+
+            if (contentItem == null || contentItem.ContentItemId == null)
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Contents/Filters/BuildDisplayFilter.cs]: Why do you test `ContentItemId`?

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Contents/Filters/BuildDisplayFilter.cs]: If the input is a `JObject` but which not represents a `ContentItem`, then a `ContentItem` is still created, but where properties are equal to their default value or null (unless `.Data and .Content` which get some values).

So, to prevent it from failling, we check one of these properties which is null in this case.

Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Contents/Filters/BuildDisplayFilter.cs]: `if (contentItem?.ContentItemId == null)`

Also please add a comment for why you test `null` here, I will forget ;)

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Contents/Filters/BuildDisplayFilter.cs]: Okay

**Comment by jtkech** [comment]
@sebastienros let me know if it is worth to do the same for filters which have a `ContentItem` input. So we could use them with a `JObject` input, e.g with the `display_text` filter.
